### PR TITLE
USBPORT_InvalidateEndpointHandler(): Properly enumerate endpoints

### PR DESCRIPTION
**Pull request by SergeGautherie**
Same fix as 085528c31ad8b91ae7af6dda9b9faa5c463ade70.

### CODE REVIEW

File: drivers/usb/usbport/endpoint.c [hunk 1]
```diff
@@ -1357,9 +1357,9 @@ USBPORT_InvalidateEndpointHandler(IN PDEVICE_OBJECT FdoDevice,
     {
         KeAcquireSpinLock(&FdoExtension->EndpointListSpinLock, &OldIrql);
 
-        Entry = &FdoExtension->EndpointList;
-
-        while (Entry && Entry != &FdoExtension->EndpointList)
+        for (Entry = FdoExtension->EndpointList.Flink;
+             Entry && Entry != &FdoExtension->EndpointList;
+             Entry = endpoint->EndpointLink.Flink)
```
Comment (ThFabba) [inline drivers/usb/usbport/endpoint.c]: Can you make this `Entry = Entry->Flink` please? Doesn't feel right using `endpoint` here when that only gets initialized below.

Comment (SergeGautherie) [reply drivers/usb/usbport/endpoint.c]: Simplified ;-)

File: drivers/usb/usbport/endpoint.c [hunk 2]
```diff
@@ -1357,9 +1357,9 @@ USBPORT_InvalidateEndpointHandler(IN PDEVICE_OBJECT FdoDevice,
     {
         KeAcquireSpinLock(&FdoExtension->EndpointListSpinLock, &OldIrql);
 
-        Entry = &FdoExtension->EndpointList;
-
-        while (Entry && Entry != &FdoExtension->EndpointList)
+        for (Entry = FdoExtension->EndpointList.Flink;
+             Entry && Entry != &FdoExtension->EndpointList;
```
Comment (SergeGautherie) [inline drivers/usb/usbport/endpoint.c]: Not this PR, but: is the `NULL` check actually needed?

**Comment by SergeGautherie** [comment]
> SergeGautherie requested a review from ThFabba now

(Wrong click...)

File: drivers/usb/usbport/endpoint.c (hunk 2 above)
Comment (Extravert-ir) [reply drivers/usb/usbport/endpoint.c]: Yeah, it should probably be changed to an `ASSERT`

Comment (SergeGautherie) [reply drivers/usb/usbport/endpoint.c]: Noted, as a future follow-up.

**Pull request merged**